web/frontend/src/api/navigation.ts

```diff
@@ -33,44 +38,55 @@ export const setWaypoint = (client: Client, lat: number, lng: number, name: stri
   rcLogConditionally(request);
 
   return new Promise((resolve, reject) => {
-    client.navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) =>
+    get(client).navigationService.addWaypoint(request, new grpc.Metadata(), (error, response) =>
       (error ? reject(error) : resolve(response)));
```

Comment:
[Nit, not new code] In general, I think we should avoid triggering side-effects in ternaries. It can be pretty tough to reason with while reading, especially combined with the callback hell of `new Promise`

Reply:
Any recs for an alternative? an if / else block or something else?

Reply:
Yeah if/else block. FWIW this will error out on the common lint config because it's a return statement from the `new Promise` constructor, which the linter will yell at you about because it's unused

Reply:
Fixed!